core/src/main/scala/org/http4s/util/UrlCoding.scala

```diff
@@ -91,7 +117,10 @@ private[util] trait UrlCodingUtils {
             in.position(mark+1)
           }
         } else {
-          while(in.hasRemaining) in.get() // just burn the rest of the bytes
+          // This is an invalid encoding. Fail gracefully by treating the '%' as
```

Comment:
This behavior was in the original spec, but didn't run because of multiple assertions in a single immutable example.  This code descends from [rl](https://github.com/backchatio/rl), which was more interested in fast normalization than RFC lawyering.  That said, I think Postel's Law applies here.